src/Altinn.Platform/Altinn.Platform.Authorization/Authorization/Services/Implementation/PolicyRetrievalPoint.cs

```diff
@@ -13,14 +17,107 @@ namespace Altinn.Platform.Authorization.Services.Implementation
+        public PolicyRetrievalPoint(IPolicyRepository policyRepository)
+        {
+            _repository = policyRepository;
+        }
+
+        /// <inheritdoc/>
+        public async Task<XacmlPolicy> GetPolicyAsync(XacmlContextRequest request)
+        {
+            string policyPath = GetPolicyPath(request);
+            Stream policyStream = await _repository.GetPolicyAsync(policyPath);
+
+            return (policyStream != null) ? ParsePolicy(policyStream) : null; 
+        }
+
+        /// <inheritdoc/>
+        public Task<bool> WritePolicyAsync(string org, string app, Stream fileStream)
         {
-            throw new NotImplementedException();
+            if (string.IsNullOrWhiteSpace(org))
+            {
+                throw new ArgumentException("Org can not be null or empty");
+            }
+
+            if (string.IsNullOrWhiteSpace(app))
+            {
+                throw new ArgumentException("App can not be null or empty");
+            }
+
+            if (fileStream == null)
+            {
+                throw new ArgumentException("The policy file can not be null");
+            }
+
+            return WritePolicyInternalAsync(org, app, fileStream);
+        }
+
+        private async Task<bool> WritePolicyInternalAsync(string org, string app, Stream fileStream)
+        {
+            string filePath = GetAltinnAppsPolicyPath(org, app);
+            return await _repository.WritePolicyAsync(filePath, fileStream);
+        }
+
+        private string GetPolicyPath(XacmlContextRequest request)
+        {
+            string org = string.Empty;
+            string app = string.Empty;
+
+            foreach (XacmlContextAttributes attr in request.Attributes)
+            {
+                if (attr.Category.OriginalString.Equals(XacmlConstants.MatchAttributeCategory.Resource))
+                {
+                    foreach (XacmlAttribute asd in attr.Attributes)
+                    {
+                        if (asd.AttributeId.OriginalString.Equals(orgAttributeId))
+                        {
+                            org = asd.AttributeValues.FirstOrDefault().Value;
+                        }
+
+                        if (asd.AttributeId.OriginalString.Equals(appAttributeId))
+                        {
+                            app = asd.AttributeValues.FirstOrDefault().Value;
+                        }
+                    }
+                }
+            }
+
+            return GetAltinnAppsPolicyPath(org, app);
+        }
+
+        private string GetAltinnAppsPolicyPath(string org, string app)
+        {
+            if (string.IsNullOrEmpty(org))
+            {
+                throw new ArgumentException("Org was not defined");
+            }
+
+            if (string.IsNullOrEmpty(app))
+            {
+                throw new ArgumentException("App was not defined");
+            }
+
+            return $"{org}/{app}/policy.xacml";
```

Comment:
The variables "org" and "app" must be sanitized by using for example the method AsFileName() on each of the variables used.

Reply:
I have now added the methode AsFileName for both org and app